routes.go

```diff
@@ -30,15 +30,17 @@ type Server struct {
 	mux  *http.ServeMux
 	db   *DB
 	addr string
+	in   chan []byte
 }
 
 // NewServer creates a new instance of the application, configured with the
 // provided addr, API roots and database handle.
-func NewServer(addr string, apiroots []string, db *DB) (*Server, error) {
+func NewServer(addr string, apiroots []string, db *DB, msgChannel chan []byte) (*Server, error) {
```

Comment:
You're probably fine naming `msgChannel` something simple like `ch` here. Its name does not last longer than the initialization of the new `Server`.